drivers/sensor/lis2dux12/lis2dux12.c

```diff
@@ -0,0 +1,386 @@
+				     CONFIG_SENSOR_INIT_PRIORITY, &lis2dux12_driver_api);
+
+/*
+ * Instantiation macros used when a device is on a SPI bus.
+ */
+
+#ifdef CONFIG_LIS2DUX12_TRIGGER
+#define LIS2DUX12_CFG_IRQ(inst)                                                                    \
+	.trig_enabled = true, .int1_gpio = GPIO_DT_SPEC_INST_GET_OR(inst, int1_gpios, {0}),        \
+	.int2_gpio = GPIO_DT_SPEC_INST_GET_OR(inst, int2_gpios, {0}),
+#else
+#define LIS2DUX12_CFG_IRQ(inst)
+#endif /* CONFIG_LIS2DUX12_TRIGGER */
+
+#define LIS2DUX12_SPI_OPERATION                                                                    \
+	(SPI_WORD_SET(8) | SPI_OP_MODE_MASTER | SPI_MODE_CPOL | SPI_MODE_CPHA)
+
+#define LIS2DUX12_CONFIG_SPI(inst)                                                                 \
+	{                                                                                          \
+		.ctx =                                                                             \
+			{                                                                          \
+				.read_reg = (stmdev_read_ptr)stmemsc_spi_read,                     \
+				.write_reg = (stmdev_write_ptr)stmemsc_spi_write,                  \
+				.mdelay = (stmdev_mdelay_ptr)stmemsc_mdelay,                       \
+				.handle = (void *)&lis2dux12_config_##inst.stmemsc_cfg,            \
+			},                                                                         \
+		.stmemsc_cfg =                                                                     \
+			{                                                                          \
+				.spi = SPI_DT_SPEC_INST_GET(inst, LIS2DUX12_SPI_OPERATION, 0),     \
+			},                                                                         \
+		.range = DT_INST_PROP(inst, range), .pm = DT_INST_PROP(inst, power_mode),          \
+		.odr = DT_INST_PROP(inst, odr),                                                    \
+		IF_ENABLED(UTIL_OR(DT_INST_NODE_HAS_PROP(inst, int1_gpio),                         \
+				   DT_INST_NODE_HAS_PROP(inst, int2_gpio)),                        \
+			   (LIS2DUX12_CFG_IRQ(inst)))                                              \
+	}
+
+/*
+ * Instantiation macros used when a device is on an I2C bus.
+ */
+
+#define LIS2DUX12_CONFIG_I2C(inst)                                                                 \
+	{                                                                                          \
+		.ctx =                                                                             \
+			{                                                                          \
+				.read_reg = (stmdev_read_ptr)stmemsc_i2c_read,                     \
+				.write_reg = (stmdev_write_ptr)stmemsc_i2c_write,                  \
+				.mdelay = (stmdev_mdelay_ptr)stmemsc_mdelay,                       \
+				.handle = (void *)&lis2dux12_config_##inst.stmemsc_cfg,            \
+			},                                                                         \
```

Comment:
Pls use STMEMSC_CTX_I2C() instead of this 4 assignments. See other STMEMSC drivers.

Reply:
Fixed